common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -284,7 +293,9 @@ static MergedShuffleFileManager newMergedShuffleFileManagerInstance(TransportCon
       // will also need the transport configuration.
       return mergeManagerSubClazz.getConstructor(TransportConf.class).newInstance(conf);
     } catch (Exception e) {
-      logger.error("Unable to create an instance of {}", mergeManagerImplClassName);
+      if(logger != null) {
+        logger.error("Unable to create an instance of {}", mergeManagerImplClassName);
+      }
```

Comment:
When can this be `null` ?

Reply:
This can be null we use this for running a few tests in the `YarnShuffleServiceSuite`.

Reply:
Oh wait, this is a static method - logger wont be available here.